src/services/account.service.ts

```diff
@@ -0,0 +1,77 @@
+import {Injectable} from '@angular/core';
+import {HttpClient} from '@angular/common/http';
+import {environment} from '../environments/environment';
+import {ServerTokenMessage} from '../model/server-token-message';
+import {ServerMessage} from '../model/server-message';
+
+const EMAIL_REGEX: RegExp = /^.+@gatech.edu$/i;
+const MIN_PASSWORD_LENGTH = 6;
+const TOKEN_NAME = 'ACCESS_TOKEN';
+const COULD_NOT_CONNECT = 'Could not connect to server.';
+
+
+@Injectable()
+export class AccountService {
+
+  isLoggedIn: boolean;
+
+  static getEmailRegex(): RegExp {
+    return EMAIL_REGEX;
+  }
+
+  static getMinPasswordLength(): number {
+    return MIN_PASSWORD_LENGTH;
+  }
+
+  constructor(private http: HttpClient) {
+    this.isLoggedIn = false;
+  }
+
+  logout(): void {
+    // TODO find some way to invalidate token?
```

Comment:
The token has a timeout after which it is no good, so it probably can be invalidated. Have you looked at the jsonwebtoken module to see if it has this functionality?

Reply:
Yeah, I've manually set the timeout for seven days (in the back end code). I don't think there is any way to invalidate tokens with jsonwebtoken. You can set the expiration time on token creation and have a max age when validating tokens, but there is no invalidate function.

Reply:
I see, so the expiration date is probably part of the hashed value?

I think I remember my team at CareerBuilder having a database with active tokens with a valid field. We could do something like that?

Reply:
Yeah, that was one idea I saw on Stack Overflow. We'll work it out in a later PR.